### PR TITLE
add duties columns

### PR DESCRIPTION
**Pull request by HanamaruSS**

### CODE REVIEW

**Comment by donjar** [comment]
Hmm, apparently Rails only creates references. Can you create foreign keys as well? This is to maintain more database integrity :)

http://guides.rubyonrails.org/active_record_migrations.html#foreign-keys

When there's a foreign key there's no need to index the columns so can just remove them!

**Comment by HanamaruSS** [comment]
so add foreign key and then remove what index?

**Comment by donjar** [comment]
If u check the schema.rb you can see that some indexes are added (those
that have relationships wrth other models). Just remove them and replace
with foreign key relationships.

**Comment by HanamaruSS** [comment]
so i remove the index for references but i keep those for unique?
btw,time_ranges start time and end time is not not-null yet

**Comment by donjar** [comment]
Retain the unique indexes. Only the indexes in the "blabla_id"

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -5,13 +5,20 @@
...
+#  matric_number :string           not null
+#  phone_number  :string           not null
+#  cell          :string           not null
+#  position      :string           not null
+#
+# Indexes
+#
+#  index_users_on_email          (email) UNIQUE
+#  index_users_on_matric_number  (matric_number) UNIQUE
+#  index_users_on_phone_number   (phone_number) UNIQUE
```
Comment (HanamaruSS) [inline spec/models/user_spec.rb]: no!
for our purposes no one should have the same contact no!
;p

File: spec/factories/sequences.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+FactoryGirl.define do
+  sequence :unique do |n|
+    name = 'aaaaaaaa'
+    n.times { name.succ! }
+    name
+  end
+  sequence :unique_int do |n|
+    int = n % 7
```
Comment (emer7) [inline spec/factories/sequences.rb]: For day testing purpose.

File: spec/factories/sequences.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+    int
+  end
+  sequence :unique_bool do |n|
+    out = if n.even?
```
Comment (emer7) [inline spec/factories/sequences.rb]: Just the same (right?)

File: spec/factories/sequences.rb (hunk 2 above)
Comment (HanamaruSS) [inline spec/factories/sequences.rb]: to generate random day

File: spec/models/duty_spec.rb [hunk 4]
```diff
@@ -2,13 +2,36 @@
...
+  end
+  it 'does not save when timeslot_id is null' do
+    duty = build(:duty, timeslot_id: nil)
+    expect(duty.save).to be false
+  end
+  it 'saves when user_id is null' do
+    duty = build(:duty, user_id: nil)
+    expect(duty.save).to be true
+  end
 end
```
Comment (HanamaruSS) [inline spec/models/duty_spec.rb]: Idea.reflect_on_association(:person).macro.should  eq(:belongs_to)

Comment (HanamaruSS) [inline spec/models/duty_spec.rb]: `expect(User.reflect_on_association(:timeslot).macro).to be :belongs_to`

found the correct one

**Pull request merged**